pkg/kudoctl/cmd/params/parser.go

```diff
@@ -51,35 +51,42 @@ func getParamsFromFiles(fs afero.Fs, filePaths []string, errs []string) (map[str
 			errs = append(errs, fmt.Sprintf("error reading from parameter file %s: %v", filePath, err))
 			continue
 		}
-		data := make(map[string]interface{})
-		err = yaml.Unmarshal(rawData, &data)
+
+		errs = GetParametersFromFile(filePath, rawData, errs, parameters)
+
+	}
+	return parameters, errs
+}
+
+func GetParametersFromFile(filePath string, bytes []byte, errs []string, parameters map[string]string) []string {
```

Comment:
If `[]string` is used to return multiple errors, why not use `[]error` instead?
Also, what's the reasoning behind the `errs` input value? It's only used for `append` calls and hence unnecessary. If it should be appended to an existing list of errors, it's much clearer if callers run
`errs = append(errs, GetParametersFromFile()...)`.

Reply:
I think the reason it was built like this is for more concise error handling. See, for example, the entry method [GetParameterMap](https://github.com/kudobuilder/kudo/pull/1541/files#diff-bf67adbf5dcb565289e52dec5b2bcfd5R22). Instead of:
```
	var errs []string

	paramsFromCmdline, errs := getParamsFromCmdline(raw, errs)
	paramsFromFiles, errs := getParamsFromFiles(fs, filePaths, errs)
```

you would end up with smth. like:

```
	var errs []string

	paramsFromCmdline, errs := getParamsFromCmdline(raw)
	if len(errs) != 0 {
		errs = append(errs, errs...)
	}
	paramsFromFiles, errs := getParamsFromFiles(fs, filePaths)
	if len(errs) != 0 {
		errs = append(errs, errs...)
	}
```

and you'll have to do it in all helper methods in this parser. I might be wrong though /cc @porridge

Reply:
FTR, this was done when changing [this code](https://github.com/kudobuilder/kudo/pull/1410/files#diff-aa194d72fb484841a8ab88b1fecbe800L11-L25) into [this](https://github.com/kudobuilder/kudo/pull/1410/files#diff-bf67adbf5dcb565289e52dec5b2bcfd5R19-R26).

@nfnt you cannot just ` errs = append(errs, GetParametersFromFile()...).` because it returns two values, not just the errors.

TBH I'm not sure why I did it this way. But comparing to @zen-dog 's example which I think should be written as follows (no need for the checks I guess, but the variable names need to vary), I think passing `errs` in as an input variable too is less error-prone and easier to read. Not a hard opinion, though.

```
	var errs []string

	paramsFromCmdline, errsFromCmdline := getParamsFromCmdline(raw)
	errs = append(errs, errsFromCmdline...)
	paramsFromFiles, errsFromFiles := getParamsFromFiles(fs, filePaths)
	errs = append(errs, errsFromFiles...)
```

Reply:
Your example looks great, that's what I had in mind.

Reply:
I thought `nil` array expansion wouldn't work but it does. Ok, I'll adjust